src/Kernel.EFSupport/Provider/IRepository.cs

```diff
@@ -0,0 +1,18 @@
+using LT.DigitalOffice.Kernel.Attributes;
+using Microsoft.AspNetCore.JsonPatch;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.Kernel.EFSupport.Provider;
+
+[AutoInject]
+public interface IRepository<T> where T : class
+{
+  Task<IQueryable<T>> GetAllAsync();
+  Task<T> GetAsync(Guid id);
+  Task<Guid> CreateAsync(T entity);
+  Task<T> EditAsync(JsonPatchDocument<T> patch);
+  Task<T> UpdateAsync(T entity);
+  Task<bool> DeleteAsync(T entity);
```

Comment:
Why not DeleteAsync(Guid id)?

Reply:
you will have to get it anyway so the idea is that you pass entity that was received from Get methods

Reply:
got it